cloud/amazon/ec2_vpc_endpoint.py

```diff
@@ -0,0 +1,347 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+module: ec2_vpc_endpoint
+short_description: Create and delete AWS VPC Endpoints.
+description:
+  - Creates AWS VPC endpoints.
+  - Deletes AWS VPC endpoints.
+  - This module support check mode.
+version_added: "2.2"
+requirements: [ boto3 ]
+options:
+  vpc_id:
+    description:
+      - Required when creating a VPC endpoint.
+    required: false
+  service:
+    description:
+      - An AWS supported vpc endpoint service. Use the ec2_vpc_endpoint_facts
+        module to describe the supported endpoint services.
+      - Required when creating an endpoint.
+    required: false
+  policy:
```

Comment:
A simpler implementation available in 2.2 is making this `type=json` in which case it will take a data structure or formatted JSON, for files you can use the lookup plugins.

Since this is a module the file needs to exist on the 'target' system, which leads to confusion when users don't execute this locally (jumphost/management host), which lookups help to avoid.